template/module/tests/test_abstract_something.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Copyright <YEAR(S)> <AUTHOR(S)>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import mock
+
+from openerp import models
+from openerp.tests.common import TransactionCase
+
+
+class AbstractSomethingTester(models.Model):
+    """ It provides a real model object to test the abstract with """
+    _name = 'abstract.something.tester'
+    _description = 'Abstract Something Tester'
+    _inherit = 'abstract.something'
+
+
+class TestAbstractSomething(TransactionCase):
+
+    @classmethod
+    def _init_test_model(cls, model_cls):
+        """ It builds a model from model_cls in order to test abstract models
+
+        Note that this does not actually create a table in the database, so
+        there may be some unidentified edge cases.
+
+        Args:
+            model_cls: (openerp.models.BaseModel) Class of model to initialize
+        Returns:
+            Model instance
+        """
+        registry = cls.env.registry
+        cr = cls.env.cr
+        inst = model_cls._build_model(registry, cr)
+        model = cls.env[model_cls._name].with_context(todo=[])
+        model._prepare_setup()
+        model._setup_base(partial=False)
+        model._setup_fields(partial=False)
+        model._setup_complete()
+        model._auto_init()
```

Comment:
This is the method that actually creates a table for the model, and it will break the isolation provided by `TransactionCase`: https://github.com/odoo/odoo/blob/18456b9d42769a522adf86183746b49735e9d8b3/odoo/models.py#L2535. Unfortunately, I don't see a way to proceed beyond the `_setup_complete()` step without introducing this issue, which defeats the entire purpose of building a model in this way.